src/components/NavigatorSearch/modes/orders.ts

```diff
@@ -17,31 +17,30 @@ export function getGqlOrderId(orderNumber: string): string {
 }
 
 function getOrdersModeActions(
-  query: string,
   intl: IntlShape,
   navigate: UseNavigatorResult,
-  order: CheckIfOrderExistsQuery["order"],
+  orders: QuickOrderSearchResult,
 ): QuickSearchAction[] {
-  const gqlId = getGqlOrderId(query);
-
-  if (isQueryValidOrderNumber(query) && maybe(() => order.id === gqlId)) {
-    return [
-      {
-        extraInfo: transformOrderStatus(order.status, intl).localized,
-        label: intl.formatMessage(messages.goToOrder, {
-          orderNumber: query,
-        }),
-        onClick: () => {
-          navigate(orderUrl(gqlId));
-
-          return false;
-        },
-        type: "action",
-      },
-    ];
+  if (!orders) {
+    return [];
   }
 
-  return [];
+  return orders.map(order => {
+    const orderNumber = order?.number ?? "";
+
+    return {
+      extraInfo: transformOrderStatus(order.status, intl).localized,
+      label: intl.formatMessage(messages.goToOrder, {
+        orderNumber,
+      }),
+      onClick: () => {
+        navigate(orderUrl(order.id));
```

Comment:
can't entries be a links? navigation based on onClick are hacky a bit

Reply:
I can improve it in the scope of the general Navigator improvements ticket (it's in backlog)